File: server/src/main/java/ru/andrey/sensor/temperaturesensor/model/Scale.java
package ru.andrey.sensor.temperaturesensor.model;

public enum Scale {
    F {
        @Override
        public double toCelsius(double fahrenheit) {
            return (fahrenheit - 32) * (5.0 / 9.0);
        }
    }, C {
        @Override
        public double toCelsius(double celcius) {
            return celcius;
        }
    }, K {
        @Override
        public double toCelsius(double kelvin) {
            return kelvin - 273.15;
        }
    };

    public abstract double toCelsius(double temperature);
}

File: server/src/main/java/ru/andrey/sensor/temperaturesensor/model/Scale.kt
package ru.andrey.sensor.temperaturesensor.model

enum class Scale {
    F {
        override fun toCelsius(temperature: Double): Double {
            return (temperature - 32) * (5.0 / 9.0)
        }
    },
    C {
        override fun toCelsius(temperature: Double): Double {
            return temperature
        }
    },
    K {
        override fun toCelsius(temperature: Double): Double {
            return temperature - 273.15
        }
    };

    abstract fun toCelsius(temperature: Double): Double
}

